pkg/controller/add_cluster.go

```diff
@@ -119,34 +111,35 @@ func (c *Controller) EnsureNamespace(cluster *myspec.M3DBCluster) (bool, error)
 func (c *Controller) ensureServices(cluster *myspec.M3DBCluster) error {
 	// TODO(schallert): support updating service spec, not sure if this only
 	// handles creation.
-	m3dbSvc, found := c.getServiceConfig(_M3DBSvcName, cluster.Spec.ServiceConfigurations)
-	if !found {
-		err := errors.New("m3db service not found in spec")
-		c.logger.Error(err.Error())
-		c.recorder.Event(cluster, corev1.EventTypeWarning, "NoServiceConfig", "could not create m3db service, missing service config in spec")
-		return err
-	}
 
-	if err := c.k8sclient.EnsureService(cluster, m3dbSvc); err != nil {
-		err = fmt.Errorf("error creating m3db service: %v", err)
-		c.logger.Error(err.Error())
-		c.recorder.Event(cluster, corev1.EventTypeWarning, "ServiceCreateError", err.Error())
-		return err
+	services := []*corev1.Service{}
+	if len(cluster.Spec.Services) != 0 {
+		services = cluster.Spec.Services
+	} else {
+		coordSvc, err := k8sops.GenerateCoordinatorService(cluster.Name)
+		if err != nil {
+			return err
+		}
+		services = append(services,
+			coordSvc,
+		)
 	}
 
-	coordSvc, found := c.getServiceConfig(_M3CoordinatorSvcName, cluster.Spec.ServiceConfigurations)
-	if !found {
-		err := errors.New("coordinator service not found in spec")
-		c.logger.Error(err.Error())
-		c.recorder.Event(cluster, corev1.EventTypeWarning, "NoServiceConfig", "could not create m3coordinator service, missing service config in spec")
+	m3dbSvc, err := k8sops.GenerateM3DBService(cluster.Name)
```

Comment:
Would this conflict if the user specifies a m3db like service within the spec? Furthermore shouldn't we allow users to pick any port configuration they wish too?